batchproducer/batchproducer.go

```diff
@@ -28,6 +30,15 @@ type Producer interface {
 	// that it’s critical to check the return value because the BatchProducer could have
 	// died in the background due to a panic (or something).
 	Add(data []byte, partitionKey string) error
+
+	// Flush stops the Producer using Stop and attempts to send all buffered records to Kinesis as
+	// fast as possible with batches of size 500 (the maximum). It blocks until either all records
+	// are sent or the timeout expires. It returns the number of records still remaining in the
+	// buffer or (possibly) an error. (It doesn’t currently return errors but that is in the
```

Comment:
:nail_care: 

Since we're thinking about future-proofing, it might make sense to add records flushed.

```go
Flush(...) (sent int, remaining int, err error)
```